idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,635 @@
+    protected static String EQUAL = "=";
+    protected static String JAR_CHECKSUMS = "jar-checksums";
+    protected static String SF_CLASSPATH = "sf-classpath";
+    protected static String TEST_CLASSES = "test-classes";
+    private static final String TARGET = "target";
+
+    /**
+     * The directory in which to store STARTS artifacts that are needed between runs.
+     */
+    protected String artifactsDir;
+
+    protected String startsDir;
+
+    protected String ekstaziDir;
+
+    protected ClassLoader loader;
+
+    protected List<Pair> jarCheckSums = null;
+
+    protected Set<String> selectedTests;
+
+    // the dependency map from test classes to their dependencies (classes)
+    protected Map<String, Set<String>> transitiveClosure;
+
+    // the dependency map from classes to their dependencies (test classes)
+    protected Map<String, Set<String>> reverseTransitiveClosure;
+
+    private Classpath sureFireClassPath;
+
+    protected boolean selectMore;
+
+    protected boolean detectOrNot;
+
+    protected boolean selectAll;
+
+    protected Path ekstaziSelectedTestsPath;
+
+    protected String ekstaziDependenciesFile;
+
+    protected Path startsSelectedTestsPath;
+
+    protected Path startsDependenciesPath;
+
+    protected boolean isEkstazi;
+
+    private Set<String> affectedTestClasses;
+
+    private static Set<String> immutableList;
+
+    @Override
+    public void execute() {
+        superExecute();
+
+        final ErrorLogger logger = new ErrorLogger();
+        this.outputPath = PathManager.detectionResults();
+        this.coordinates = mavenProject.getGroupId() + ":" + mavenProject.getArtifactId() + ":" + mavenProject.getVersion();
+
+        logger.runAndLogError(() -> defineSettings(logger, mavenProject));
+        if (this.runner == null) {
+            return;
+        }
+        long startTime = System.currentTimeMillis();
+        try {
+            affectedTestClasses = computeAffectedTests(mavenProject);
+            if (!detectOrNot) {
+                System.out.println("Not detect flaky tests at the first run");
+                return;
+            }
+        } catch (IOException | MojoExecutionException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        timing(startTime);
+
+        startTime = System.currentTimeMillis();
+        logger.runAndLogError(() -> detectorExecute(logger, mavenProject, moduleRounds(coordinates)));
+        timing(startTime);
+    }
+
+    // TODO: make Starts and Ekstazi's deps similar
+    private Set<String> computeAffectedTests(final MavenProject project) throws IOException, MojoExecutionException, ClassNotFoundException {
+        Set<String> affectedTests = new HashSet<>();
+        Set<String> allTests = new HashSet<>(getTestClasses(project, this.runner.framework()));
+
+        if (isEkstazi) {
+            selectedTests = getSelectedTests(ekstaziSelectedTestsPath);
+            checkSelectAll();
```

Comment:
Probably need some comment to explain why it is necessary to check this specifically for Ekstazi.